x-pack/plugins/security_solution/public/detections/components/host_isolation/use_host_isolation_action.tsx

```diff
@@ -129,20 +121,11 @@ export const useHostIsolationAction = ({
 
   const isIsolationActionDisabled = useMemo(() => {
     if (sentinelOneManualHostActionsEnabled && isSentinelOneAlert) {
-      return (
-        !sentinelOneAgentStatus ||
-        sentinelOneAgentStatus?.isUninstalled ||
-        sentinelOneAgentStatus?.isPendingUninstall
-      );
+      return !agentStatus || agentStatus?.isUninstalled || agentStatus?.isPendingUninstall;
```

Comment:
Ok. so this I think explain my question around why we included `*uninstall*` type of info. 

The API should be agent type agnostic and should not be returning `agentType` specific data. Lets discuss if you still feel it should be there. Maybe there is something else we can use (ex. agent `status` or if we somehow keep `capabilities`, then maybe we calculate this for S1 in the API and add the value to `capabilities`... maybe... not sure if there would be other repercussions )

Reply:
I agree but see my [comment](https://github.com/elastic/kibana/pull/178625#discussion_r1553652086). I assume that SentinelOne isolate/release are a bit more restricted than endpoint (where we don't care if the agent is uninstalled, the action will fail eventually in that case) and maybe it goes into a pending state when a sentinel one agent is uninstalled/pending uninstall?

Reply:
Ok. I see. I only coded that way for `8.13` because that was the way it was coded by before. I don't know why it was done that way, but: if we don't do this check for endpoint, we probably should not do it for sentinelone either.

Lets not introduce additional properties in this API's response to capture `status` of the agent. [I looked at `HostStatus` type](https://github.com/elastic/kibana/blob/6ecbe53736919419b87635cfc603e7cbaeb1a3f2/x-pack/plugins/security_solution/common/endpoint/types/index.ts#L438-L471) and we already have values there for `unenrolled` and for `updating` - which seem to be documented as meaning the same thing as `uninstalled` and `uninstalling`. 

Can you make sure that the S1 agent status values are mapped to correctly to our own `status` values and then use that in the UI (if necessary - I still say that if we don't do the check for Endpoint, lets not do it for SentinelOne or other agent types)?

Reply:
done 1db3d63abef5e53ee95a556e4a5fb9e8c332bf24